openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/TypeGeneratorUtils.java

```diff
@@ -255,8 +256,9 @@ private static RecordFieldWithDefaultValueNode getRecordFieldWithDefaultValueNod
                 defaultValueToken = AbstractNodeFactory.createIdentifierToken("\"" +
                         fieldSchema.getDefault().toString() + "\"");
             }
-        } else if (!defaultValue.matches("^[0-9]*$") &&
-                !(defaultValue.startsWith("[") && defaultValue.endsWith("]"))) {
+        } else if (!defaultValue.matches("^[0-9]*$") && !defaultValue.matches("^(\\d*\\.)?\\d+$")
```

Comment:
can't simplify the double negatives in here? 
```suggestion
        } else if (defaultValue.matches("[0-9]*$") && defaultValue.matches("(\\d*\\.)?\\d+$")
```

Reply:
We can't simplify the double negative here since one is for capturing integer numbers and the other for capturing decimals and float. Therefore given input should be fulfilled these two regexes to become strings. If it is the string we suppose to add `"` to the default value since swagger didn't provide value within `"`otherwise we give the value as it is.